src/Adapter/Product/Combination/Repository/CombinationMultiShopRepository.php

```diff
@@ -397,23 +408,32 @@ public function deleteByProductId(ProductId $productId, ShopConstraint $shopCons
      *
      * @return CombinationId[]
      */
-    public function getCombinationIds(ProductId $productId, ShopConstraint $shopConstraint): array
-    {
+    public function getCombinationIds(
+        ProductId $productId,
+        ShopConstraint $shopConstraint,
+        ?ProductCombinationFilters $filters = null
```

Comment:
not entirely sure about this, it smells abit.
We already require shop_id in filters and then we also have shopConstraint and they are kinda both used to determine the shop. (so one of them feels redundant).
We also use the shop_id in the searchQueryBuilder, so it would seem like we could just have the filters (and require shop_id inside them) without using shopConstraint in this case :thinking: (query builder also uses shop_id from filters and it has strict interface, so we cannot really add additional ShopConstraint param there and I don't think it would be worth it).
Maybe we could build the ShopConstraint inside the filters? Or have it nullable, so that when shop_id = null we would use Constraint::AllShops? (we will have a case for specific prices, where we will need to get combinations for all shops too.).
wdyt @jolelievre

Reply:
Oh I think I should be able to use your ShopCriteriaInterface once its merged from this PR https://github.com/PrestaShop/PrestaShop/pull/30495

Reply:
@zuk3975 the PR just got merged so you should be able to rebase yours Also now that the shop constraint can be built by using `ShopFilters` as a base I think `ProductCombinationFilters` should extend `ShopFilters` and I also think it shouldn't handle the shopId in its generated filterId anymore
Actually it was probably overkill in the first place since the `admin_filter` table already has a shop column,